lib/import/utility/extraction_utilities.rb

```diff
@@ -0,0 +1,152 @@
+require 'csv'
+# Common location for useful functions and constants
+module Import
+  module Utility
+    # Extraction utilities code
+    module ExtractionUtilities
+      # Common utility methods
+      class CommonUtility
+        # TODO: this pile of regexs is a mess and needs to be cleaned up
+        GENE_ONLY = '(?<location>c\.[^ \.]+)'.freeze
+        GENE_LOC = "#{GENE_ONLY} ?(?<protein>\(p\.[^\)\(]*\))?".freeze
+        BRCA_BASE = 'br?c?a?'.freeze
+        BRCA     = BRCA_BASE + '(?<brca>1|2)'.freeze
+        EXON_LOC = 'exons? (?<exons>\d+[a-z0-9]*(?:-\d+[a-z0-9]*)?)'.freeze
+        VARIANT  = '(?:familial)?(?<variantclass>(?: likely) pathogenic)?' \
+        ' BRCA(?<brca>1|2) (?:mutation|sequence variant)'.freeze
+        ZYGOSITY = '(?<zygosity>[a-z]+)zygous'.freeze
+        GENE_LOCATION = '(?<location>c\.[^ \.]+) ?(?<protein>\(p\.[^)]*\))?'.freeze
+        EXON_LOCATION = 'exons? (?<exons>\d+[a-z0-9]*(?:-\d+[a-z0-9]*)?)'.freeze
+        LOCATION_REGEX = /(#{GENE_LOCATION}(?: in (?:brca(?:1|2) )?#{EXON_LOCATION})?|#{EXON_LOCATION})/i
+
+        ZYGOSITY_REGEX = /(?<zygosity>[a-z]+)zygous/
+
+        def extract_single_mutation(report_string, genotype)
+          return false if report_string.nil?
+
+          if report_string.scan(LOCATION_REGEX).chunk { |x| x }.map(&:first).size == 1
+            matches = report_string.match(LOCATION_REGEX)
+            genotype.add_gene_location(matches[:location])
+            genotype.add_protein_impact(matches[:protein])
+            genotype.add_exon_location(matches[:exons])
+            if report_string.scan(ZYGOSITY_REGEX).size == 1
+              matches = report_string.match(ZYGOSITY_REGEX)
+              genotype.add_zygosity(matches[:zygosity])
+            end
+            true
+          else
+            false # TODO: clean up
+          end
+        end
+
+        def extract_brca(string_list)
+          string_list.
+            reject(&:nil?).
+            map { |entry| entry.scan(/br?c?a?(?<brca>1|2)/i).map { |match| match[0] } }.
+            flatten.
+            reject(&:nil?).
+            map(&:to_i).
+            sort.
+            chunk { |n| n }.
+            map(&:first)
+        end
+      end
+
+      # TODO: the accuracy of this extractor could be notably improved if it can be converted
+      #       to interpret the formal grammar of HGSV nomenclature, as several HGSV python
+      #       packages do. This applies mostly to standarising c. and p. forms and catching
+      #       some edge cases/unusual mutations which would currently be rejected.
+
+      # Provide utilities to extract c., p., and exon-based location information from
+      # raw strings
+      class LocationExtractor < CommonUtility
+        CDNA = '\\*?\\d+((_|-|\\+)\\d+)? ?([a-z]+(?:>|<)[a-z]+|(del|dup|ins)' \
+        ' ?(?:[atcg]*|[0-9]*))*(?=(?:\\s|$|\\()|,)'.freeze
+
+        def initialize(write_csv = false)
```

Comment:
Please ue kwargs when defining method with boolean argument - [example here](https://stackoverflow.com/questions/67391214/style-optionalbooleanparameter-use-keyword-arguments-when-defining-method-with)

Remember to change all calls of this method